ARE/components/actuator.tooltip/src/main/java/eu/asterics/component/actuator/tooltip/GUI.java

```diff
@@ -0,0 +1,232 @@
+    private int screenHeight;
+
+    /**
+     * The class constructor, initialises the GUI
+     *
+     * @param owner
+     *            the owner class instance
+     */
+    public GUI(final TooltipInstance owner) {
+        super("TooltipPanel");
+
+        this.owner = owner;
+        GraphicsDevice gd = GraphicsEnvironment.getLocalGraphicsEnvironment().getDefaultScreenDevice();
+        screenWidth = gd.getDisplayMode().getWidth();
+        screenHeight = gd.getDisplayMode().getHeight();
+
+        setUndecorated(true);
+        setBackground(new Color(0, 0, 0, 0)); // transparent !
+        setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+        setOpacity(0.5f);
+    }
+
+    /**
+     * Return currently active Tooltip filename.
+     * 
+     * @return
+     */
+    String getTooltipFilename() {
+        return actImageFileName;
+    }
+
+    /**
+     * Tell if Tooltip is currently active (visible)
+     * 
+     * @return
+     */
+    boolean tooltipsActive() {
+        return tooltipActive;
+    }
+
+    /**
+     * Activates the current Tooltip image. Sends an etpTooltipActivated afterwards.
+     */
+    void activateTooltips() {
+        if (actTooltip < 0) {
+            actTooltip = owner.propTooltipStartIndex;
+        }
+        loadImage(actTooltip);
+        owner.etpTooltipActivated.raiseEvent();
+        this.tooltipActive = true;
+    }
+
+    /**
+     * Deactivates the current Tooltip image. Sends an etpTooltipDeactivated afterwards.
+     */
+    void deactivateTooltips() {
+        resetImage();
+        owner.etpTooltipDeactivated.raiseEvent();
+        showTooltip(false);
+        this.tooltipActive = false;
+    }
+
+    /**
+     * Selects the next index of Tooltip images. If the @see {@link #TOOLTIP_INDEX_MAX} is reached, starts with {@link #TOOLTIP_INDEX_MIN}.
+     */
+    void navigateNextTooltip() {
+        actTooltip = (actTooltip + 1 > TOOLTIP_INDEX_MAX) ? TOOLTIP_INDEX_MIN : actTooltip + 1;
+    }
+
+    /**
+     * Selects the previous index of Tooltip images. If the @see {@link #TOOLTIP_INDEX_MIN} is reached, starts with {@link #TOOLTIP_INDEX_MAX}. *
+     */
+    void navigatePreviousTooltip() {
+        actTooltip = (actTooltip - 1 < TOOLTIP_INDEX_MIN) ? TOOLTIP_INDEX_MAX : actTooltip - 1;
+    }
+
+    /**
+     * Sets the location where the Tooltip should be shown.
+     * 
+     * @param x
+     * @param y
+     */
+    void setMouseXY(float x, float y) {
+        mouseX = (float) sanitizeValue(x, 0, screenWidth);
+        mouseY = (float) sanitizeValue(y, 0, screenHeight);
+    }
+
+    /**
+     * Paints the Tooltip image at the defined location or at the mouse cursor location, if not set.
+     */
+    @Override
+    public void paint(Graphics g) {
+        // If the mouseX and mouseY values have been set from outside, use them, otherwise default to current mouse position.
+        if (mouseX < 0 || mouseY < 0) {
+            setLocation(MouseInfo.getPointerInfo().getLocation());
```

Comment:
I think we should not use the mouse position here, but the `propInitialX` and `propInitialY` properties. Maybe `mouseX` and `mouseY` therefore is a bad name for the variables, the intention is/was that the tooltip plugin should be not natively related to the mouse position but only in connection with the Mouse plugin.